src/cpp/configuration/DiscoveryServerParticipantConfiguration.cpp

```diff
@@ -88,32 +94,41 @@ std::map<std::string, std::string> DiscoveryServerParticipantConfiguration::tls_
             throw ConfigurationException("TLS configuration must be of map yaml type or empty");
         }
 
-        std::vector<const char*> tls_tags = {
-            TLS_CA_TAG,
-            TLS_PASSWORD_TAG,
-            TLS_PRIVATE_KEY_TAG,
-            TLS_CERT_TAG,
-            TLS_DHPARAMS_TAG
+        std::vector<std::pair<const char*, std::string*>> tls_tags = {
+            {TLS_CA_TAG, &certificate_authority_file},
+            {TLS_PASSWORD_TAG, &private_key_file_password},
+            {TLS_PRIVATE_KEY_TAG, &private_key_file},
+            {TLS_CERT_TAG, &certificate_chain_file},
+            {TLS_DHPARAMS_TAG, &dh_params_file},
         };
 
         for (auto tls_tag : tls_tags)
         {
-            if (raw_configuration_[TLS_TAG][tls_tag])
+            if (raw_configuration_[TLS_TAG][tls_tag.first])
             {
                 try
                 {
-                    result.insert({tls_tag, raw_configuration_[TLS_TAG][tls_tag].as<std::string>()});
+                    (*tls_tag.second) = raw_configuration_[TLS_TAG][tls_tag.first].as<std::string>();
```

Comment:
It would be nice to check here if the given file exists for the entries which correspond to paths.

Reply:
Could be done in Yaml module (#129 )